Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py

```diff
@@ -85,7 +97,29 @@ def process(ccdb: Ccdb, object_path: str, delay: int,  from_timestamp: int, to_t
             logger.debug(f"     not in the allowed period, skip this bucket")
             preservation_list.extend(run_versions)
         else:
-            logger.debug(f"     not in the grace period")
+            logger.debug(f"    not in the grace period")
+
+            if delete_first_last:
+                logger.debug(f"    delete_first_last is set")
+                run_versions.sort(key=lambda x: x.createdAt)
+                # Get flag cleaner_2nd from first object (if there)
+                cleaner_2nd = "cleaner_2nd" in run_versions[0].metadata
+                if cleaner_2nd or len(run_versions) < 4:
+                    logger.debug(f"        first version has flag cleaner_2nd or there are less than 4 version, "
+                                 f"we continue to next bucket")
+                    preservation_list.extend(run_versions)
+                    continue
+                # flag second with `cleaner_2nd`
+                ccdb.updateValidity(run_versions[1], run_versions[1].validFrom, run_versions[1].validTo,
```

Comment:
@knopers8 I think that here we should not actually update the validity right ?

Reply:
now we probably never should update validity

Reply:
ah wait, my bad, this is actually not touching the validity but adds the flag.

Reply:
@knopers8 is it ok ? it is not touching the validity, it is just using this method to add the flag.

Reply:
yes, but perhaps we should have another method for doing this.

Reply:
I have created a dedicated method. Behind the scene we still have to specify the validity due to the way the CCDB rest api works.

Reply:
thank you